packages/radio-group/src/RadioGroup.tsx

```diff
@@ -84,90 +59,89 @@ interface RadioGroupState {
  * @param props.className classname applied to RadioGroup container.
  * @param props.name Name passed to each Radio belonging to the RadioGroup.
  * @param props.variant Variant to determine if component will appear `default` or `light`.
+ * @param props.size Determines the size of the Radio components Can be 'small' or 'default.
  */
-export default class RadioGroup extends PureComponent<
-  RadioGroupProps,
-  RadioGroupState
-> {
-  static displayName = 'RadioGroup';
-
-  static propTypes = {
-    variant: PropTypes.oneOf(['default', 'light']),
-    className: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    onChange: PropTypes.func,
-    children: PropTypes.node,
-    name: PropTypes.string,
-  };
-
-  static defaultProps = {
-    variant: Variant.Default,
-    className: '',
-    onChange: () => {},
-  };
-
-  state: RadioGroupState = {
-    value: null,
-  };
+function RadioGroup({
+  variant = Variant.Default,
+  size = Size.Default,
+  className,
+  onChange,
+  children,
+  value: controlledValue,
+  name: nameProp,
+}: RadioGroupProps) {
+  const isControlled = controlledValue != null;
+
+  let defaultChecked = '';
+
+  !isControlled &&
```

Comment:
We should use a traditional `if` block here.

Reply:
👍